mks/model/stuf_02_04.py

```diff
@@ -353,11 +352,48 @@ def extract_address(persoon_tree: Tag, is_amsterdammer):
         {"name": "officieleStraatnaam", "parser": to_string},
     ]
 
-    addresses = persoon_tree.find_all("PRSADRINS")
```

Comment:
Had je niet gewoon iets als dit kunnen doen om de structuur zoveel mogelijk hetzelfde te laten?

```
    addresses_cor = persoon_tree.find_all("PRSADRCOR")
    addresses_vbl = persoon_tree.find_all("PRSADRVBL")

    def add_type(address_type, addresses):
        return [{ **address, "type": address_type } for address in addresses]

    if is_nil(addresses_cor):
        addresses_cor = []

    if is_nil(addresses_vbl):
        addresses_vbl = []

    if not addresses_cor and not addresses_vbl:
        return {}, []

    addresses = add_type(addresses_cor, 'cor') + add_type(addresses_vbl, 'vbl')
```

Reply:
Volgens mij heb je niet meer aanpassingen nodig in dit model dan.

Reply:
@timvanoostrom Ik kan het object niet spreaden zoals jij voorsteld. Dan krijg ik een error, volgens mij omdat het geen object is maar een xml structuur? Ik snap niet helemaal hoe het zit, maar werkt het bij jou wel?

Reply:
Ah, ja het is een Bs4 Element. Check. Ik heb even gekeken. Volgens mij moet dit werken:

```
addresses_cor = persoon_tree.find_all("PRSADRCOR")
    addresses_vbl = persoon_tree.find_all("PRSADRVBL")

    def add_types(addresses, address_type):
        def add_type(address):
            address["adresType"] = address_type
            return address

        return [add_type(address) for address in addresses]

    if is_nil(addresses_cor):
        addresses_cor = []

    if is_nil(addresses_vbl):
        addresses_vbl = []

    if not addresses_cor and not addresses_vbl:
        return {}, []

    addresses = add_types(addresses_cor, "correspondentie") + add_types(
        addresses_vbl, "woon"
    )
```